### PR TITLE
Feature/on hover

### PR DESCRIPTION
**Pull request by lerning**

### CODE REVIEW

File: examples/range.js [hunk 1]
```diff
@@ -161,7 +161,7 @@ ReactDOM.render(
   <div>
     <div style={style}>
       <p>Basic Range，`allowCross=false`</p>
-      <Range allowCross={false} defaultValue={[0, 20]} onChange={log} />
+      <Range allowCross={false} defaultValue={[0, 20]} onChange={log} onMouseOver={log} onMouseLeave={logu}/>
```
Comment (SimonRuppGreene) [inline examples/range.js]: i assume the onMouseLeave is supposed to be `log`?

File: src/Range.jsx [hunk 2]
```diff
@@ -62,7 +62,12 @@ class Range extends React.Component {
 
     const { bounds } = this.state;
     const value = nextProps.value || bounds;
-    const nextBounds = value.map((v, i) => this.trimAlignValue(v, i, nextProps));
+
+    if (value.length % 2 !== 0){
+      return
+    }
+
+    const nextBounds = value
```
Comment (SimonRuppGreene) [inline src/Range.jsx:70]: i assume this is doing something with not setting state if the incoming range has an odd number of items, but i'm not totally clear on the necessity of that

File: src/Range.jsx [hunk 3]
```diff
@@ -117,12 +123,16 @@ class Range extends React.Component {
       recent: this.prevMovedHandleIndex,
     });
 
-    const prevValue = bounds[this.prevMovedHandleIndex];
-    if (value === prevValue) return;
+    return
```
Comment (SimonRuppGreene) [inline src/Range.jsx]: what's this for?

File: src/Range.jsx (hunk 2 above)
Comment (lerning) [reply src/Range.jsx:70]: sometimes, due to annoying logic mainly regarding the ability to join ranges, an array/range comes in with 1,3,5, etc, which gets filtered out by the above logic

File: src/Range.jsx (hunk 3 above)
Comment (lerning) [reply src/Range.jsx]: absolutely nothing

**Pull request merged**